sgqlc/types/__init__.py

```diff
@@ -1398,13 +1400,18 @@ class Enum(BaseType, metaclass=EnumMeta):
     >>> print(json.dumps(Fruits.__to_json_value__(Fruits.APPLE)))
     "APPLE"
 
+    >>> print(json.dumps(Fruits(Variable('input'))))
+    "$input"
+
     '''
     __kind__ = 'enum'
     __choices__ = ()
 
     def __new__(cls, json_data, selection_list=None):
         if json_data is None:
             return None
+        if isinstance(json_data, Variable):
```

Comment:
i think this is better done in another level, before it tries to create the enum. Check the backtrace when you get the error and go one level up, something like `__to_graphql...`

Reply:
In the traceback, the nearest `__to_graphql_input__` is for a more complex value/type: 
```
value = {'field': $orderByField, 'direction': $orderByDirection}
```
From there on down, it's just `Input.__init__`, `ContainerType.__init__`, `ContainerType.__populate_field_data`...